Consul + Vault - ACLs (#194)
hashicorp/katakoda: consul-secure-acl-vault/step3.md

> ### Create Consul server policy

Reviewer: ```suggestion
### Create a Consul server policy
```

> Once the ACL system is initialized, if you check Consul logs again you should find some warnings in your logs:

Reviewer: ```suggestion
Once the ACL system is initialized, you should still find some warnings in your Consul logs:
```

`cat ~/log/consul.log`{{execute T1}}

```
 [WARN]  agent: Node info update blocked by ACLs: node=c19d6af9-760b-a3cc-bbdd-b4f7209a79de accessorID=00000000-0000-0000-0000-000000000002
 [WARN]  agent: Coordinate update blocked by ACLs: accessorID=00000000-0000-0000-0000-000000000002
```

> These warnings are ACL logs from which you can infer the following: 

Reviewer: ```suggestion
These warnings indicate that: 
```

> - one node is trying to update its info in your Consul datacenter and gets denied access by the ACL system.

Reviewer: ```suggestion
- a node is trying to update its info in your Consul datacenter,  but denied by the ACL system because it does not have the correct permissions.
```

> - the node is not using any token to perform the request (because `00000000-0000-0000-0000-000000000002` is the anonymous policy used when no token is presented)

Reviewer: ```suggestion
- the agent does not have an assigned token to perform the request (because `00000000-0000-0000-0000-000000000002` is the anonymous policy used when no token is presented)
```

> First steps towards a more fine grained ACL configuration is to create tokens for the server nodes so that they could interact properly with the rest of the Consul datacenter, without assigning them an administrative token.

Reviewer: ```suggestion
The first step towards a more fine grained ACL approach is to create individual tokens for the server agents, so that they can interact properly with the rest of the Consul datacenter, without assigning them an administrative token.
```

> For the scope of this lab you are going to create a policy that will permit server nodes to register themselves using a name template of "server-<count_or_id>" and has permissions to locate other nodes and resolve services.

Reviewer: ```suggestion
For this lab, you will create a policy that permits server agents to register themselves, using a name template of "server-<count_or_id>", and have the permissions to locate other agents and discover services.
```

 [WARN]  agent: Coordinate update blocked by ACLs: accessorID=00000000-0000-0000-0000-000000000002
```

These warnings indicate that: 

> - the agent is trying to update its info in your Consul datacenter, but denied by the ACL system because it does not have the correct permissions.

Reviewer 2: ```suggestion
- the agent is trying to update its info in your Consul datacenter, but has been denied by the ACL system because it does not have the correct permissions.
```

- the agent does not have an assigned token to perform the request (because `00000000-0000-0000-0000-000000000002` is the anonymous policy used when no token is presented)

> The first step towards a more fine grained ACL approach is to create individual tokens for the server agents, so that they can interact properly with the rest of the Consul datacenter, without assigning them an administrative token.

Reviewer 2: ```suggestion
The first step towards a more fine grained ACL approach is to create individual tokens for the server agents so that they can interact properly with the rest of the Consul datacenter without being assigned an administrative token.
```

> For this lab, you will create a policy that permits server agents to register themselves, using a name template of "server-<count_or_id>", and have the permissions to locate other agents and discover services.

Reviewer 2: ```suggestion
For this lab, you will create a policy that permits server agents to register themselves. The servers will follow the naming pattern `server-<count_or_id>`, and have the necessary permissions to locate other agents and discover services.
```
